packages/components/src/popover/index.tsx

```diff
@@ -287,11 +287,45 @@ const UnforwardedPopover = (
 				? undefined
 				: normalizedPlacementFromProps,
 		middleware,
-		whileElementsMounted: ( referenceParam, floatingParam, updateParam ) =>
-			autoUpdate( referenceParam, floatingParam, updateParam, {
-				layoutShift: false,
-				animationFrame: true,
-			} ),
+		whileElementsMounted: (
+			referenceParam,
+			floatingParam,
+			updateParam
+		) => {
+			const cleanupAutoUpdate = autoUpdate(
+				referenceParam,
+				floatingParam,
+				updateParam,
+				{
+					layoutShift: false,
+					animationFrame: true,
+				}
+			);
+
+			// Observe mutations on the parent div of the inserter to update toolbar position.
+			const targetElement = document.querySelector(
+				'.interface-interface-skeleton__body'
+			);
```

Comment:
Usually, we shouldn't work directly with DOM. Have you considered using a ref instead?